waku/v2/node/wakunode2.nim

```diff
@@ -480,7 +480,13 @@ proc mountRelay*(node: WakuNode,
   
   info "mounting relay", rlnRelayEnabled=rlnRelayEnabled, relayMessages=relayMessages
 
-  node.switch.mount(wakuRelay)
+  proc relayMatch(proto: string): bool {.gcsafe.} =
+    ## Matches the WakuRelayCodec with any postfix.
+    ## E.g. if WakuRelayCodec is `/vac/waku/relay/2.0.0` it matches:
+    ## `/vac/waku/relay/2.0.0`, `/vac/waku/relay/2.0.0-beta2`, `/vac/waku/relay/2.0.0-actualnonsense`
+    return proto.startsWith(WakuRelayCodec)
```

Comment:
This makes sense to me conceptually, nice! Seems like something we can work with going forward with different versions and behavior etc.

Reply:
Yeah, the idea is for the matcher to be extended to something more sophisticated in future. :)